docs/src/components/ComponentDoc/ComponentExample/ComponentExample.tsx

```diff
@@ -59,73 +63,117 @@ const childrenStyle: React.CSSProperties = {
  * Allows toggling the the raw `code` code block.
  */
 class ComponentExample extends React.Component<ComponentExampleProps, ComponentExampleState> {
-  anchorName: string
   kebabExamplePath: string
 
-  constructor(props) {
-    super(props)
+  static getClearedActiveState = () => ({
+    showCode: false,
+    showRtl: false,
+    showVariables: false,
+    showTransparent: false,
+  })
 
-    const { examplePath } = props
+  static getAnchorName = props => examplePathToHash(props.examplePath)
 
-    this.anchorName = examplePathToHash(examplePath)
-    this.state = {
-      showCode: this.isActiveHash(),
-      componentVariables: {},
-      usedVariables: {},
-      showRtl: examplePath && examplePath.endsWith('rtl'),
-      showTransparent: false,
-      showVariables: false,
-      wasEverVisible: false,
+  static isActiveHash = props => {
+    const anchorName = ComponentExample.getAnchorName(props)
+    const formattedHash = getFormattedHash(props.location.hash)
+
+    return anchorName === formattedHash
+  }
+
+  static getStateFromURL = props => {
+    return qs.parse(props.location.search, {
+      ignoreQueryPrefix: true,
+      decoder: (raw, parse) => {
+        const result = parse(raw)
+        return result === 'false' ? false : result === 'true' ? true : result
```

Comment:
```suggestion
        return result === 'false' ? false : (result === 'true' ? true : result)
```
To make it easier to read

Reply:
Heads up, code style is not configurable in our repo as it is dictated by `prettier`. You can see what happens when you add the parenthesis [here](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEBHAzgOgA4EMBO6cAFNvhNlgDYRi4wCW0mRBYAFgDQAEwAOlG7cGAcygR8cAIoBXOPgCeABUkAzBgA8k3GPjmcBQgCZxIJ-NuL5cAdx55CcAJTcAvAD5ehod0hR0MNyS6DJUga7cDkRWtk7eQpIwMviCwaHhrhEA5Kq4VERZ3AD83Ln5cNyWaWFumdxZunKFJY0V2tUwcYLcAL4GUD1dIJwgFIzQ6MigBOQ2SgQIkyi4AG4QDEbDIABG1mAA1nAwAMp4YAxQIsitI+wwALZUAOrsDPDoZ3DHi28MK28KZDgdCTEYXIj4GAqXAie64ZBlIgjABW6A0ACE9ocTrh7nAADIXOAIvJIkCojTHC4iKjSGQQeAk8ojKLyIHbXDbBQ0KBbMgXGBPDYwdjIAAcAAYWeQiE9rNggWQ4BCVsSRqh6fAVBQliBcOgALRQOBwEybEaSDUMSTQ2HwpCIuAjIj3BhMsnoam02QM4kO0lOkAwTlCowi5AAJhGulwDCo1IAwhB7nCgeJjVsZEQACqcpaOno9IA).